detectors/service/service_test.go

```diff
@@ -0,0 +1,50 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package service
+
+import (
+	"context"
+	"regexp"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestDetectFormat(t *testing.T) {
+	detector := New()
+
+	serviceResource, err := detector.Detect(context.Background())
+	assert.NoError(t, err)
+
+	var uuid string
+
+	for _, kv := range serviceResource.Attributes() {
+		if kv.Key == "service.instance.id" {
+			uuid = kv.Value.AsString()
+		}
+	}
+
+	matched, err := regexp.MatchString("^[A-Fa-f0-9]{8}-[A-Fa-f0-9]{4}-[A-Fa-f0-9]{4}-[A-Fa-f0-9]{4}-[A-Fa-f0-9]{12}$", uuid)
```

Comment:
tip for the future: you can add `/i` at the end of the regex to make it case insensitive, this way you don't need to add `A-F` and `a-f`, which have the risk of forgetting one of them